c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -1820,36 +1787,33 @@ class DeviceCachingAllocator {
   // on every stream, so the block's previous lifetime ends before any new
   // lifetime begins. This check relies solely on the DAG topology and does not
   // require event queries, making it safe to use during capture.
-  //
-  // This function iterates over all deferred blocks, determines if their empty
-  // nodes are reusable according to the above criteria, and frees the block if
-  // so.
   void free_safe_blocks_in_capture(
       const std::shared_ptr<GatheredContext>& context,
       cudaStream_t stream) {
-    auto reusable_empty_nodes = get_reusable_empty_nodes(stream);
+    auto info = stream_get_capture_info(stream);
 
     // If there are no reusable empty nodes (e.g., not currently capturing),
     // there is nothing to do.
-    if (reusable_empty_nodes.empty()) {
+    if (info.status == cudaStreamCaptureStatusNone || info.num_terminals == 0) {
       return;
     }
+    auto& graph_context = graph_reuse_context[info.graph];
```

Comment:
so entries in the `graph_reuse_context` are leaked and never deleted? Would it also prevent freeing resources associated with the graph when the user wants to destroy a graph?

Reply:
I’ve added a map cleanup [here](https://github.com/pytorch/pytorch/pull/162186/files#diff-acc6337586bf9cdcf0a684380779300ec171897d05b8569bf439820dc8c93bd5R2489) to ensure the cached info is deleted once capture ends.